infinispan-remote/encoding/src/main/java/org/infinispan/tutorial/simple/encoding/InfinispanEncodingCaches.java

```diff
@@ -0,0 +1,34 @@
+package org.infinispan.tutorial.simple.encoding;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.commons.marshall.UTF8StringMarshaller;
+import org.infinispan.tutorial.simple.connect.Infinispan;
+
+import java.net.URI;
+
+public class InfinispanEncodingCaches {
+
+   public static void main(String[] args) throws Exception {
+      ConfigurationBuilder builder = Infinispan.connectionConfig();
+
+      URI textCacheURI = InfinispanEncodingCaches.class.getClassLoader().getResource("textCache.xml").toURI();
+
+      builder.remoteCache("textCache")
+            .configurationURI(textCacheURI).marshaller(new UTF8StringMarshaller());
+
+      RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build());
+      RemoteCache<String, Object> textCache = cacheManager.getCache("textCache");
+
+      System.out.println("== Cache with text encoding and text marshaller.");
+      // Put an integer
```

Comment:
It is a bit strange to create a text cache and store integers. Maybe you could use phrases instead, with different charsets (chinese, ascii, etc)?

Reply:
the idea is to show that even if I try an integer in java it is stored as string. but I'm more than open to suggestions for this example

Reply:
ok, no prob. In any case, ```text/plain``` is not stored as a String, it should be stored as a ```byte[]```, associated with a ```charset```, e.g. ```text/plain; charset=us-ascii``` (default is ```UTF-8```)